src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkBase.cs

```diff
@@ -10,22 +10,21 @@ namespace Xamarin.Android.Tools
 	abstract class AndroidSdkBase
 	{
 		string[]? allAndroidSdks;
-		string[]? allAndroidNdks;
 
 		public string[] AllAndroidSdks {
 			get {
-				if (allAndroidSdks == null)
-					allAndroidSdks = GetAllAvailableAndroidSdks ().Distinct ().ToArray ();
+				if (allAndroidSdks == null) {
+					var dirs = new List<string?> ();
+					dirs.Add (AndroidSdkPath);
+					dirs.AddRange (GetAllAvailableAndroidSdks ());
+					allAndroidSdks = dirs.Where (d => ValidateAndroidSdkLocation (d))
+						.Select (d => d!)
+						.Distinct ()
```

Comment:
@jonpryor will `Distinct()` lose the ordering again? or does the ordering not matter here?

Reply:
Yes, it will lose the ordering again, but this is only from `AndroidSdkInfo.AllAndroidSdks`, which "nobody" should be using, and which *always* had `.Distinct()` semantics, so anybody using that method shouldn't see anything bizarre.

I hope.

Reply:
Confirmed we're only using GetAllAvailableAndroidSdks (and also every time we use it we're using Distinct too :)